CourseApp/Program/FileCreator.cs

```diff
@@ -0,0 +1,42 @@
+namespace CourseApp.Program
+{
+    using System;
+    using System.Collections.Generic;
+
+    public class FileCreator
+    {
+        public FileCreator(string filename, string extension, string weight, string weightModificator)
+        {
+            FileName = filename;
+            Extension = extension;
+            Weight = weight;
+            WeightModificator = weightModificator;
+        }
+
+        public FileCreator(List<string> customValues)
+        {
+            if (customValues.Count != 4)
+            {
+                throw new NullReferenceException();
```

Comment:
Hm , i guess it's may be far from null - you may throw simply general Exception - or, create own exception